Source/Core/DolphinQt/TAS/BalanceBoardWidget.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QDialog>
+
+#include "Common/CommonTypes.h"
+
+class BalanceBoardWidget : public QWidget
+{
+  Q_OBJECT
+public:
+  explicit BalanceBoardWidget(QWidget* parent);
+
+signals:
+  void ChangedX(u16 x);
+  void ChangedY(u16 y);
+
+public slots:
+  void SetX(u16 x);
+  void SetY(u16 y);
+
+protected:
+  void paintEvent(QPaintEvent* event) override;
+  void mousePressEvent(QMouseEvent* event) override;
+  void mouseMoveEvent(QMouseEvent* event) override;
+  void handleMouseEvent(QMouseEvent* event);
+
+private:
+  u16 m_x = 0;
+  u16 m_y = 0;
+  bool m_ignore_movement = false;
+};
+
+// Should be part of class but fails to compile on mac os
+static const u16 balance_range = 1000;
```

Comment:
What's the compilation error?

Reply:
Copy-paste from `IRWidget`:

https://github.com/dolphin-emu/dolphin/blob/a9e04e0a72e59ee0cb9e0daf017561b2fb4b93dc/Source/Core/DolphinQt/TAS/IRWidget.h#L37-L39

I'm not sure what it actually is.

Reply:
You shouldn't be copy/pasting things with caveats if you don't actually know what the reason is or if it actually applies in this scenario.

Reply:
I did a blame on that, and the original commit is 5fe72700fade70511852979af33d8da5506754fc from #6359 which points to https://buildbot.dolphin-emu.org/#/builders/33/builds/337 for the build failure.

And, yeah, that's something I don't want to have once this is done, but for the moment I left it in.  (I feel like it would be worse to get rid of the comment in the duplicated code &mdash; ideally it won't be super duplicated in the final version anyways.)

Reply:
That error is likely due to the fact they were used in std::min/std::max (which takes arguments by reference), so for this to work pre-c++17, the following would need to be done;

```cpp
// In header
class Thing
{
private:
  static const int value = 9;
};

// In cpp file:
int Thing::value;
```

However, in C++17 this is no longer applicable, as `static inline` or `static constexpr` will make a static class member's declaration also act as its definition, so it would be sufficient in this case to make the variable `constexpr` instead of `const` for it to be placed within the class.